msteams-platform/tabs/how-to/app-initialization-process.md

```diff
@@ -0,0 +1,91 @@
+---
+title: App Initialization Process for Tab Apps
+description: Learn about the app initialization process, including the APIs involved and recommendations for their use in code.
+author: JoshuaPartlow
+ms.author: joshuapa
+ms.service: msteams
+ms.topic: concept-article #Required; leave this attribute/value as-is.
+ms.date: 03/16/2025
+
+#CustomerIntent: As a developer, I want to understand how the app initialization process works for tab apps so that my app can make the appropriate API calls and work as recommended.
+---
+
+# App initialization process
+
+App initialization is a foundational aspect of authoring a tab app and ensures that apps work as expected. All apps created for Microsoft Teams that use TeamsJS must be initialized. The process of app initialization is a finite state machine and it uses APIs with specific roles to advance and monitor the event flow. The following sections will:
+
+- Review the APIs for app initialization and their role
+- Discuss the types of app initialization flows
+- Show and explain the state changes during app initialization
+- Provide recommendations for app initialization
+
+## APIs for app initialization
+
+There are specific APIs within the [Teams JavaScript client library](using-teams-client-library.md) (TeamsJS) used during the app initialization process. While their specific use during initialization is discussed in later sections, at a high level they're:
+
+- [app.initialize()](/javascript/api/@microsoft/teams-js/app#@microsoft-teams-js-app-initialize) - Initializes the TeamsJS library. Before any other library calls are made, `app.initialize()` has to be called and complete successfully.
+- [app.notifyApploaded()](/javascript/api/@microsoft/teams-js/app#@microsoft-teams-js-app-notifyapploaded) - Notifies the frame that the app is partially loaded and that the loading indicator can be hidden if one was shown.
+- [app.notifySuccess()](/javascript/api/@microsoft/teams-js/app#@microsoft-teams-js-app-notifysuccess) - Notifies the frame that the app is fully loaded and was successfully initialized. It's now ready for user interaction.
```

Comment:
I think it is good to say `app.notifySuccess` could clear loading indicator as well.

Reply:
added small addition in next rev